hw30/index.js

```diff
@@ -0,0 +1,223 @@
+                            listItem.setSelected(true);
+                            if (e.target === listItem.element) {
+                                break;
+                            }
+                        }
+                        return;
+                    }
+
+                    const targetIndex = this.items.findIndex((listItem) => listItem.element === e.target);
+                    const currentlySelectedIndex = this.items.findIndex((listItem) => listItem.element === currentlySelected[0].element);
+                    this.clearSelected();
+                    const [start, end] = targetIndex < currentlySelectedIndex ? [targetIndex, currentlySelectedIndex] : [currentlySelectedIndex, targetIndex];
+
+                    for (let index = start; index <= end; index++) {
+                        this.items[index].setSelected(true);
+                    }
+                    return;
+                }
+
+                this.clearSelected();
+                const selected = this.getListItem(e.target);
+                selected.setSelected(true);
+            });
+
+        }
+
+        clearSelected() {
+            this.items.forEach((elem) => {
+                elem.setSelected(false);
+            });
+        }
+
+        getSelected() {
+            const selectedItems = this.items.filter((listItem) => listItem.isSelected);
+            return selectedItems.length ? selectedItems : null;
+        }
+
+        getListItem(target) {
+            return this.items.find((listItem) => listItem.element === target);
+        }
+
+        addListItem(text, position) {
+            const listItem = new ListItem(text);
+            const listItemElement = this.createListElement(listItem);
+
+            if (position === 'start') {
+                this.items.unshift(listItem);
+                this.element.insertAdjacentElement('afterbegin', listItemElement);
+            } else {
+                this.items.push(listItem);
+                this.element.insertAdjacentElement('beforeend', listItemElement);
+            }
+            return listItem;
+        }
+
+        removeListItem(listItem) {
+            listItem.remove();
+            this.items = this.items.filter((item) => item !== listItem);
+        }
+
+        createListElement(item) {
+            const listItemElement = document.createElement('li');
+            listItemElement.innerText = item.text;
+            item.setElement(listItemElement);
+            return listItemElement;
+        }
+
+        renderList(container) {
+            const list = this.element;
+            container.appendChild(list);
+        }
+
+        get listItems() {
+            return this.items;
+        }
+    }
+
+    class Menu {
+        constructor(elem, taskInput) {
+            this.input = taskInput;
+            this._elem = elem;
+            elem.onclick = this.onClick.bind(this);
+        }
+
+        validateTask() {
+            const value = this.input.value;
```

Comment:
actually this wasn't in the task. But ok, if you use user input it's a good practice to use trim() to prevent just typing blank spaces, or to trim a line that might accidentally have extra spaces in it
```javascript
const value = this.input.value.trim();
```